src/Test.UnitTests.Sarif/Core/LocationTests.cs

```diff
@@ -85,5 +92,68 @@ public void Location_LogicalLocation_WhenSetToNull_RemovesLogicalLocationsArray(
 
             location.LogicalLocations.Should().BeNull();
         }
+
+        [Fact]
+        public void Location_SerializeId()
+        {
+            var location = new Location();
+            AssertShouldSerializeId(location, false);
+
+            location.Id = -1;
+            AssertShouldSerializeId(location, false);
```

Comment:
add test for long.MinValue - 1
add test for long.MinValue
add test for int.MinValue
add test for -2

Reply:
added all